### PR TITLE
GitHub action on PR: markdownlint

### PR DESCRIPTION
**Pull request by PriceChild**
Part resolves: https://github.com/Libera-Chat/libera-chat.github.io/issues/69

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache
+        uses: actions/cache@v2.1.4
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
```
[the remark this answers is not in the document]
Comment (PriceChild) [reply .github/workflows/lint.yml:21]: @swantzter is https://github.com/PriceChild/libera-chat.github.io/commit/0f83404cbebd3a73acb36039989f900f199419ee what you are looking for?

File: content/_guides/connect.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ appear in WHOIS (a 276 numeric).
 
 Libera.Chat is also reachable via
 [Tor](https://www.torproject.org/), bound to some restrictions. You can't
-directly connect to irc.libera.chat via Tor; use the following hidden service
+directly connect to `irc.libera.chat` via Tor; use the following hidden service
```
[the remark this answers is not in the document]
Comment (PriceChild) [reply content/_guides/connect.md:59]: I put this in a code block just so the "common name" linter didn't complain about it not being capitalised which I thought was weird for a url. Yes, replacing it with `[irc.Libera.Chat](irc:irc.libera.chat)` is cool. (or keeping capitalisation & adding a rule disable)

Just below, there's the hidden service .onion address which is formatted as a code block. Do you want that as a link too?

I would very much like to make them the same (code blocks, given half already are) just to get the linting merged, then make changes like that later? You're the boss!

File: .github/workflows/lint.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (PriceChild) [reply .github/workflows/lint.yml:21]: (The reasoning for me adding either was for caching of the npm install.)

[the remark this answers is not in the document]
Comment (swantzter) [reply .github/workflows/lint.yml:21]: Yeah that works

File: content/_guides/connect.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (glguy) [reply content/_guides/connect.md:59]: If we're going to make things clickable IRC URLs, let's always do them as ones that point to TLS, however that works.

[the remark this answers is not in the document]
Comment (PriceChild) [reply content/_guides/connect.md:59]: @glguy makes sense, updated my example in comment above with `ircs`.

As discussed elsewhere though, I'd prefer to limit this PR to the minimal changes required to pass linting, using existing styles etc. If we want to change styles (e.g. moving to fenced code blocks, or linkifying all hostnames) then I feel a subsequent and separate PR would be best suited to that.

[the remark this answers is not in the document]
Comment (PriceChild) [reply content/_guides/connect.md:61]: Agreed we'll leave these as code blocks for now.

**Review by swantzter — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
+
+      - uses: xt0rted/markdownlint-problem-matcher@v1
+      - run: npx markdownlint-cli '**/*.md'
```
Comment (swantzter) [inline .github/workflows/lint.yml:29]: actually, now that we have a packkage.json we might as well move this here to make local dev easier
```suggestion
      - run: npm run markdownlint
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "libera-chat.github.io",
+  "version": "1.0.0",
+  "description": "Libera Chat website",
```
Comment (swantzter) [inline package.json:4]:
```suggestion
  "description": "Libera Chat website",
  "scripts": {
    "lint:md": "markdownlint-cli '**/*.md'",
    "lint:md:fix": "npm run lint:md -- --fix"
  },
```

**Review by swantzter — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Cache
+        uses: actions/cache@v2.1.4
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (swantzter) [inline .github/workflows/lint.yml:26]: also now taht we ahve a package.json
```suggestion
          node-version: 12
      - run: npm i --prefer-offline
```

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+name: Lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Comment (swantzter) [inline .github/workflows/lint.yml:7]:
```suggestion
on: [push, pull_request]
```

**Pull request merged**